### PR TITLE
feat: allow `maxAutomaticTokenAssociations` values to be -1

### LINKED ISSUE #709: Update HIP-904: `AccountCreateTransaction` to allow maxAutomaticTokenAssociations value to support -1 as a valid value

### PR DESCRIPTION
**Pull request by rwalworth**
This PR updates any and all `maxAutomaticTokenAssociations` variable for accounts and contracts to be -1. Tests were also added to verify correct functionality.

**Related issue(s)**:

Fixes #709 #710 #720

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by jeromy-cannon — DISMISSED** [review]

**Comment by rwalworth** [comment]
This is waiting to be tested until services `v0.52`, which allows `maxAutoTokenAssociations` to be -1, is tagged and available for testing.

**Review by gsstoykov — COMMENTED** [review]
Updated the `local-node` version in CI but the tests are still failing due to missing proto definitions from `v0.52.0`.

**Review by jeromy-cannon — APPROVED** [review]

**Comment by rwalworth** [comment]
This has been tested with `v0.52.0-alpha.6` version of `hedera-services` which contain the max auto association changes, so this should be good to review again.

**Review by gsstoykov — DISMISSED** [review]
LGTM.

File: .github/workflows/zxc-build-library.yaml [hunk 1]
```diff
@@ -107,7 +107,7 @@ jobs:
       #     run: tree /F build
 
       - name: Start the local node
-        run: sudo npx @hashgraph/hedera-local start -d --network local --network-tag=0.51.0-alpha.3
+        run: sudo npx @hashgraph/hedera-local start -d --network local --network-tag=0.52.0-alpha.4
```
Comment (gsstoykov) [inline .github/workflows/zxc-build-library.yaml:110]: `0.52.0-alpha.4` should we try with `alpha.6` before merging as you mentioned in the PR comments?

**Review by jeromy-cannon — DISMISSED** [review]

**Review by gsstoykov — APPROVED** [review]
LGTM.

**Review by jeromy-cannon — APPROVED** [review]

**Pull request merged**